### PR TITLE
[configcat] use internal route

### PR DESCRIPTION
**Pull request by akosyakov**
## Related Issue(s)

fix IDE-185

## How to test

- Check logs of all dependent components that they can resolve feature flag.
- Try to change some feature flags and see whether changes are picked up.

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - ak-ide-185</li>
	<li><b>🔗 URL</b> - <a href="https://ak-ide-185.preview.gitpod-dev.com/workspaces" target="_blank">ak-ide-185.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
</details>

/hold

### CODE REVIEW

**Review by laushinka — APPROVED** [review]

File: install/installer/pkg/common/common.go [hunk 1]
```diff
@@ -431,7 +431,7 @@ func ConfigcatEnv(ctx *RenderContext) []corev1.EnvVar {
 		},
 		{
 			Name:  "CONFIGCAT_BASE_URL",
-			Value: "https://" + ctx.Config.Domain + "/configcat",
+			Value: "http://proxy.default.svc.cluster.local:9547/configcat",
```
Comment (easyCZ) [inline install/installer/pkg/common/common.go:434]:
```suggestion
			Value: "http://" + ClusterAddress(proxy.Component, ctx.Namespace, port) 
```

Comment (akosyakov) [reply install/installer/pkg/common/common.go:434]: I will do but after testing, don't want to wait for another build.

File: components/proxy/conf/Caddyfile [hunk 2]
```diff
@@ -146,6 +146,13 @@
 	}
 }
 
+# Internal configcat endpoint
+:9547 {
```
Comment (easyCZ) [inline components/proxy/conf/Caddyfile:150]: Do we need a seperate port? could it be part of the 9546 port above?

**Review by easyCZ — APPROVED** [review]
Assuming preview works, this LGTM

File: components/proxy/conf/Caddyfile (hunk 2 above)
Comment (akosyakov) [reply components/proxy/conf/Caddyfile:150]: I thought about it but then service port has already name `analytics`. Should we rename it? Is it fine?

Comment (akosyakov) [reply components/proxy/conf/Caddyfile:150]: It also has slightly different network policies. So not sure.

**Comment by akosyakov** [comment]
I reviewed logs and played with ide service feature flags. It seems to work. I’m going to unhold.

/unhold

**Pull request merged**